packages/amplify-graphql-model-transformer/src/resources/rds-model-resource-generator.ts

```diff
@@ -0,0 +1,69 @@
+import { TransformerContextProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import { ResourceConstants } from 'graphql-transformer-common';
+import { ModelResourceGenerator } from './model-resource-generator';
+import { ModelVTLGenerator, RDSModelVTLGenerator } from '../resolvers';
+import { createRdsLambda, createRdsLambdaRole } from '../resolvers/rds';
+import { MYSQL_DB_TYPE } from '../types';
+import {RDSConnectionSecrets} from '@aws-amplify/graphql-transformer-core';
+
+export const RDS_STACK_NAME = 'RdsApiStack';
+
+/**
+ * An implementation of ModelResourceGenerator responsible for generated CloudFormation resources
+ * for models backed by an RDS data source
+ */
+export class RdsModelResourceGenerator extends ModelResourceGenerator {
+  protected readonly generatorType = 'RdsModelResourceGenerator';
+
+  generateResources(context: TransformerContextProvider): void {
+    if (this.isEnabled()) {
+      const secretEntry = context.datasourceSecretParameterLocations.get(MYSQL_DB_TYPE);
+      if (!secretEntry) {
+        throw new Error('Secret entry not found for RDS access, unable to create lambda');
+      }
+      const {
+        RDSLambdaIAMRoleLogicalID,
+        RDSLambdaLogicalID,
+        RDSLambdaDataSourceLogicalID,
+        RDSLambdaDataSourceLogicalName,
+      } = ResourceConstants.RESOURCES;
+      const lambdaRoleStack = context.stackManager.getStackFor(RDSLambdaIAMRoleLogicalID, RDS_STACK_NAME);
+      const lambdaStack = context.stackManager.getStackFor(RDSLambdaLogicalID, RDS_STACK_NAME);
+      const role = createRdsLambdaRole(
+        context.resourceHelper.generateIAMRoleName(RDSLambdaIAMRoleLogicalID),
+        lambdaRoleStack,
+        secretEntry as RDSConnectionSecrets,
+      );
+      const lambda = createRdsLambda(
+        lambdaStack,
+        context.api,
+        role,
+        {
+          username: secretEntry?.username ?? '',
+          password: secretEntry?.password ?? '',
+          host: secretEntry?.host ?? '',
+          port: secretEntry?.port ?? '',
+          database: secretEntry?.database ?? '',
+        },
```

Comment:
can we directly pass the `secretEntry as RDSConnectionSecrets` here instead of creating another object?